packages/libs/htz-react-base/scripts/build/lib.js

```diff
@@ -8,7 +8,10 @@ checkDir('dist').then(hasDist => {
   if (!hasDist) {
     fs.mkdirSync('dist');
   }
-  if (process.argv.length < 3) {
+  if (process.argv.includes('watch')) {
+    process.argv = [ process.argv[0], process.argv[1], '--watch', 'src', '--out-dir', 'dist/lib', ];
```

Comment:
I'd do this a bit differently to preserve flexibility:
 
```suggestion
    const watchIndex = process.argv.indexOf('watch');
    const const [ executer, command, ...args ] = process.argv;
    if (process.argv.length > 4) {  
      args.splice(watchindex, 1);
      process.argv = [ executer, command, '--watch', ...args ];
    }
    else {
       process.argv = [ executer, command, '--source-maps', '--watch', 'src', '--out-dir', 'dist/lib', ];
    }
```

But please check this snippet before applying it, because I haven't tested it

Reply:
The way you did it doesn't let us override default arguments like `--source-maps` when passing `watch`.

Try something like this:

```js
const { argv, } = argv;
// user did not pass any arguments
if (argv.length < 3) {
  argv.push('src', '--source-maps', '--out-dir', 'dist/lib');
}
// User passed arguments, and we need to check if `watch` is one of them
else {
  const watchIndex = argv.indexOf('watch');
  const hasWatch = watchIndex !== -1;

  if (hasWatch) {
    argv[watchindex] = '--watch';

    // User only passed the `watch` argument, add defaults
    if (argv.length < 4) {
      argv.push(
        '--source-maps',
        'src',
        '--out-dir',
        'dist/lib'
     ); 
    }
  }
}
```

But again, I didn't test this, so please validate that it works as expected